File: src/main/java/random/telegramhomebot/telegram/MockBot.java
package random.telegramhomebot.telegram;

import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;
import random.telegramhomebot.config.ProfileService;

@Slf4j
@Profile(ProfileService.MOCK_BOT)
@Component
public class MockBot implements Bot {

	@Override
	public void sendMessage(String messageText) {
		log.info(messageText);
	}
}

File: src/main/java/random/telegramhomebot/telegram/MockBot.kt
package random.telegramhomebot.telegram

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Component
import random.telegramhomebot.utils.logger

@ConditionalOnProperty(prefix = "mock-bot", value = ["enabled"], havingValue = "true")
@Component
class MockBot : Bot {
    val log = logger()
    override fun sendMessage(messageText: String) {
        log.info(messageText)
    }
}
